conversion_example.py

```diff
@@ -7,27 +7,32 @@ def train_mnist():
     mnist = tf.keras.datasets.mnist
 
     (x_train, y_train), (x_test, y_test) = mnist.load_data()
-    x_train, x_test = x_train / 255.0, x_test / 255.0
+    x_train_normed, x_test_normed = x_train / 255.0, x_test / 255.0
+
+    x_train_normed, x_test_normed = x_train_normed.reshape((-1,28,28,1)), x_test_normed.reshape((-1,28,28,1))
 
     model = tf.keras.models.Sequential([
-    tf.keras.layers.Flatten(input_shape=(28, 28)),
-    tf.keras.layers.Dense(128, activation='relu', use_bias=False),
-    tf.keras.layers.Dense(64, activation='relu', use_bias=False),
-    tf.keras.layers.Dense(10, activation='softmax',use_bias=False)
+        tf.keras.layers.Conv2D(16,5,activation='relu',use_bias=False,input_shape=(28,28,1)),
+        tf.keras.layers.AveragePooling2D(2),
+        tf.keras.layers.Conv2D(8,5,activation='relu',use_bias=False),
+        tf.keras.layers.Flatten(input_shape=(28,28,1)),
+        tf.keras.layers.Dense(128, activation='relu', use_bias=False),
+        tf.keras.layers.Dense(64, activation='relu', use_bias=False),
+        tf.keras.layers.Dense(10, activation='softmax',use_bias=False)
     ])
 
     model.compile(optimizer='adam',
                 loss='sparse_categorical_crossentropy',
                 metrics=['accuracy'])
 
-    model.fit(x_train, y_train, epochs=5)
+    model.fit(x_train_normed[:10000], y_train[:10000], epochs=5)
 
-    model.evaluate(x_test[:1000], y_test[:1000])
+    model.evaluate(x_test_normed[:1000], y_test[:1000])
 
-    return model, x_train*255., y_train, x_test*255., y_test
+    return model, x_train, y_train, x_test, y_test
 
 tf_model, x_train, y_train, x_test, y_test = train_mnist()
 
 # Create models
-relu_ann = ReLUANN()
-g_model = tg.convert_model(tf_model,relu_ann,x_test[:1000],y_test[:1000])
\ No newline at end of file
+relu_ann = ReLUANN(membrane_capacitance=0.4)
```

Comment:
I agree about your intuition regarding the membrane capacitance but it would be really nice if the user didn't have to specify it manually here **and** surely you don't want this membrane capacitance for neurons following dense layers? Maybe ``ReLUANN`` could have a ``dense_membrane_capacitance`` and ``sparse_membrane_capacitance`` constructor argument which get applied to the two sorts of layers respectively (and default to 0.4 and 1.0)

Reply:
I tried setting different capacitances for sparse and dense layers (as opposed to using a common value for all layers). There are a couple of differences resulting from this:
1. The optimal capacitance values are different (expected)
2. Changing the layers in the network has a drastic effect on the optimal capacitance values, and hence might require some amount of parameter searching. Is this acceptable or should we go back to using a single value for a network (all layers)

Reply:
Interesting! How do you mean by "changing the layers"? The number of layers, their dimensions?

Reply:
The number of layers. Changing the layer dimensions doesn't seem to have much effect

Reply:
I guess more layers will increase the variability in spike latency through the network so you would probably want a larger membrane time constant in deeper layers and therefore a larger capacitance - do the optimal capacitance values increase as you go deeper in the network?

Reply:
That makes sense! I'll try varying the capacitance values with depth and let you know.

Reply:
It is also possible that one of the weight normalisation schemes from the literature may help with this problem as they generally aim to keep the activation within the linear region of the postsynaptic neurons

Reply:
Yes. That would probably solve it.
Varying capacitance values with depth produced the expected results. Also, keeping capacitance constant, I tried increasing single_example_time to 500 and threshold voltage to -56. This gives much better results for a wider range of capacitance values

Reply:
Ok cool, maybe leave that capacitance alone for now then. Maybe you can start investigating weight normalization as part of getting a bigger CIFAR10 network working?

Reply:
Sure